src/api/l_event.c

```diff
@@ -176,15 +171,10 @@ static int l_lovrEventQuit(lua_State* L) {
   EventData data;
 
   int argType = lua_type(L, 1);
-  if (argType == LUA_TSTRING && !strcmp("restart", lua_tostring(L, 1))) {
-    data.quit.restart = true;
-    data.quit.exitCode = 0;
-    luax_checkvariant(L, 2, &data.quit.cookie);
-  } else if (argType == LUA_TNUMBER || lua_isnoneornil(L, 1)) {
-    data.quit.restart = false;
+  if (argType == LUA_TNUMBER || lua_isnoneornil(L, 1)) {
     data.quit.exitCode = luaL_optint(L, 1, 0);
   } else {
-    return luaL_argerror (L, 1, "number, nil or the exact string 'restart' expected");
+    return luaL_argerror (L, 1, "number or nil expected");
   }
```

Comment:
This ifelse could be reduced to just `luaL_optint(L, 1, 0)` since opt* will error if the thing isn't nil or the appropriate type.  Error message will probably be more canonical too.